### PR TITLE
Config

### PR DESCRIPTION
**Pull request by dhagandev**
Added in yml config.

### CODE REVIEW

File: twitter-example.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+consumerKey:
```
Comment (kev-chan) [inline twitter-example.yml]: use a twitter parent object

File: src/main/java/liuni/LiUniDriver.java [hunk 2]
```diff
@@ -1,44 +1,36 @@
...
-    }
 
-    private static void setupTwitter() {
-        KeyHandler keyHandler = new KeyHandler();
+    @Override
+    public void run(TwitterConfig config, Environment env) {
         try {
-            keyHandler.setWriter(new BufferedWriter(new FileWriter("twitter4j.properties")));
+            config.setWriter(new BufferedWriter(new FileWriter("twitter4j.properties")));
+            config.createTwitter4JProp();
```
Comment (kev-chan) [inline src/main/java/liuni/LiUniDriver.java]: do you need a twitter4j properties file?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/main/java/liuni/LiUniDriver.java [hunk 3]
```diff
@@ -1,44 +1,39 @@
...
         }
         catch (Exception e) {
             e.printStackTrace();
         }
     }
+
     @Override
-    public void run(Configuration config, Environment env) {
+    public void run(TwitterConfig config, Environment env) {
+        Twitter twitter = config.createTwitterConfig();
```
Comment (kev-chan) [inline src/main/java/liuni/LiUniDriver.java]: do you need to be configuration twitter at this level?

File: src/main/java/liuni/TwitterConfig.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+import twitter4j.Twitter;
+import twitter4j.TwitterFactory;
+import twitter4j.conf.ConfigurationBuilder;
+
+import javax.validation.constraints.NotNull;
+
+public class TwitterConfig extends io.dropwizard.Configuration {
```
Comment (kev-chan) [inline src/main/java/liuni/TwitterConfig.java]: no need to declare io.dropwizard here.  Use an import

File: src/main/java/liuni/TwitterConfig.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    @NotNull
+    private Map<String, String> twitter;
```
Comment (kev-chan) [inline src/main/java/liuni/TwitterConfig.java]: Don't use a map as then you've got all of the string constants below

File: src/main/java/liuni/TwitterConfig.java [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+    @NotEmpty
```
Comment (kev-chan) [inline src/main/java/liuni/TwitterConfig.java]: why is this marked NotEmpty?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/main/java/liuni/TwitterConfig.java (hunk 5 above)
Comment (dhagandev) [reply src/main/java/liuni/TwitterConfig.java]: I'm not sure I understand? The constants below are the key names, not the values.

File: src/main/java/liuni/TwitterConfig.java (hunk 6 above)
Comment (dhagandev) [reply src/main/java/liuni/TwitterConfig.java]: Because the key names should not be empty?

File: src/main/java/liuni/TwitterConfig.java (hunk 4 above)
Comment (dhagandev) [reply src/main/java/liuni/TwitterConfig.java]: Resolved

File: src/main/java/liuni/LiUniDriver.java (hunk 3 above)
Comment (dhagandev) [reply src/main/java/liuni/LiUniDriver.java]: Resolved

File: src/main/java/liuni/TwitterStatus.java [hunk 7]
```diff
@@ -2,16 +2,20 @@
 
 import twitter4j.Twitter;
 import twitter4j.TwitterException;
-import twitter4j.TwitterFactory;
 import twitter4j.Status;
 
 public class TwitterStatus {
 
     public static final int TWITTER_CHAR_MAX = 280;
     private static Twitter twitter;
```
Comment (kev-chan) [inline src/main/java/liuni/TwitterStatus.java]: why is Twitter static?

File: src/main/java/liuni/TwitterTimeline.java [hunk 8]
```diff
@@ -2,14 +2,18 @@
 
 import twitter4j.ResponseList;
 import twitter4j.Twitter;
-import twitter4j.TwitterFactory;
 import twitter4j.Status;
 
 public class TwitterTimeline {
     private static Twitter twitter;
```
Comment (kev-chan) [inline src/main/java/liuni/TwitterTimeline.java]: same for this one

**Review by kev-chan — APPROVED** [review]

**Pull request merged**